test/fetch/data-uri.js

```diff
@@ -1,215 +1,215 @@
-'use strict'
```

Comment:
@KhafraDev help here? these fail...

Reply:
```
\undici\lib\fetch\index.js:971
    response.body = { stream: response.body.stream.pipeThrough(transformStream) }
                                                   ^

TypeError: Cannot read properties of undefined (reading 'pipeThrough')
    at fetchFinale (\undici\lib\fetch\index.js:971:52)
    at mainFetch (\undici\lib\fetch\index.js:761:5)

Node.js v17.7.1
```

Reply:
`response.body` is a Uint8Array

Reply:
Okay this is quite an easy fix, but difficult to track down (my favorite)!

`dataURLProcessor` returns either a string or a Uint8Array
https://github.com/nodejs/undici/blob/6a0388f38da39538368eed5ef16d218f8510515c/lib/fetch/dataURL.js#L57
https://github.com/nodejs/undici/blob/6a0388f38da39538368eed5ef16d218f8510515c/lib/fetch/dataURL.js#L67

So then, possibly either a spec oversight or my mistake, the body is never extracted:
https://github.com/nodejs/undici/blob/6a0388f38da39538368eed5ef16d218f8510515c/lib/fetch/index.js#L879

Which means it's now a Uint8Array (most likely). The fix is just adding `extractBody(dataURLStruct.body)[0]`. All tests pass as well.

Reply:
@ronag

Reply:
@KhafraDev thanks! Seems like a spec bug that it's missing extract body here. @dominic

Reply:
@annevk

Reply:
This one is on us I'm pretty sure:

```
A body consists of:

*    A stream (a [ReadableStream](https://streams.spec.whatwg.org/#readablestream) object).

*    A source (null, a [byte sequence](https://infra.spec.whatwg.org/#byte-sequence), a [Blob](https://w3c.github.io/FileAPI/#dfn-Blob) object, or a [FormData](https://xhr.spec.whatwg.org/#formdata) object), initially null.

*    A length (null or an integer), initially null.
```

undici only works if the body is a stream, but it should work for all of the other types as well. `dataURLProcessor` returns a `byte sequence` body (Uint8Array in our case).